streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimestampedCacheFlushListener.java

```diff
@@ -37,41 +37,20 @@
         myNode = this.context.currentNode();
     }
 
-    @SuppressWarnings("unchecked")
-    TimestampedCacheFlushListener(final org.apache.kafka.streams.processor.ProcessorContext context) {
-        this.context = (InternalProcessorContext<KOut, Change<VOut>>) context;
-        myNode = this.context.currentNode();
-    }
-
-    @Override
-    public void apply(final KOut key,
-                      final ValueAndTimestamp<VOut> newValue,
-                      final ValueAndTimestamp<VOut> oldValue,
-                      final long timestamp) {
-        final ProcessorNode prev = context.currentNode();
-        context.setCurrentNode(myNode);
-        try {
-            context.forward(
-                key,
-                new Change<>(getValueOrNull(newValue), getValueOrNull(oldValue)),
-                To.all().withTimestamp(newValue != null ? newValue.timestamp() : timestamp));
-        } finally {
-            context.setCurrentNode(prev);
-        }
-    }
-
     @Override
     public void apply(final Record<KOut, Change<ValueAndTimestamp<VOut>>> record) {
         @SuppressWarnings("rawtypes") final ProcessorNode prev = context.currentNode();
         context.setCurrentNode(myNode);
         try {
             context.forward(
-                record.withValue(
-                    new Change<>(
-                        getValueOrNull(record.value().newValue),
-                        getValueOrNull(record.value().oldValue)
-                    )
-                )
+                record
+                    .withValue(
+                        new Change<>(
+                            getValueOrNull(record.value().newValue),
+                            getValueOrNull(record.value().oldValue)))
+                    .withTimestamp(
+                        record.value().newValue != null ? record.value().newValue.timestamp()
+                            : record.timestamp())
```

Comment:
Huh. It looks like we missed this before, eh?

Reply:
Haha, true.